habitat-lab/habitat/config/habitat/task/rearrange/cat_place.yaml

```diff
@@ -0,0 +1,47 @@
+# @package habitat.task
+
+defaults:
+  - /habitat/task: task_config_base
+  - /habitat/task/actions:
+    - arm_action
+    - base_velocity
+  - /habitat/task/measurements:
+    - robot_force
+    - force_terminate
+    - robot_colls
+    - end_effector_to_rest_distance
+    - end_effector_to_goal_distance
+    - object_to_goal_distance
```

Comment:
Dense reward to a random location on the receptacle (currently using the geometric goal). There is an option to switch this to a sparse reward.

Reply:
Using a random location on the receptacle as the goal would cause the agent to learn some spurious correlations. The correct distance to goal in this case would be the distance to the sampling volume. But with the new receptacle annotation codebase, I would expect that code to change. So we can keep using the sparse reward for now I think

Reply:
For now, I set sparse reward as True by default. 

For the dense reward, the distance would be from the agent's position to which point on the sampling volume?

- I do think the current location I am using: the GeoGoal place locations are entirely random (I will verify by plotting the goal locations in the episodes for a particular receptacle). By rewarding based on random locations, the hope was that the agent will learn to place the object at some reasonable location. But there's a possibility that rewarding based on random locations may confuse the agent on where to place. I am still testing this "place" in the latest setup and will try to understand the behaviour the agent learns.

- Alternatively, we can reward the agent based on the distance to the closest target location where the object can be safely placed. For this, we may need to take into account the size of the object in the agent's hand and also other objects on the receptacle. I can implement this. For now, I will test (with/without scale) first if one of the sparse reward or the current dense reward work.

Reply:
The second option could make sense. We can just assume that the max object size is 1 ft in each direction and just find the location which is 1ft away from the receptacle edge (while handling the case that there might be receptacles with less than 1ft of length or width)